### DISCUSSION TITLE
HTTP Endpoint Issue {System.Dynamic.ExpandoObject} with Target Type Not Being Passed to Activity Input

### DISCUSSION THREAD

**Discussion by adamfisher** [elsa-workflows/elsa-core#4232] [Q&A] [answered]
I have a workflow defined that receives an HTTP event that should then de-serialize it into my custom model and pass it on to my custom activity as in input as shown below.

![image](https://github.com/elsa-workflows/elsa-core/assets/1834711/5dc42d81-d7d5-4d84-ace3-87a1c2f60cd2)

I set the HttpEndpoint to the assembly qualified name of my model:

![image](https://github.com/elsa-workflows/elsa-core/assets/1834711/e02bf645-c181-4121-b30d-d0ad7eb10c21)

I pass the body of the **HttpEndpoint** activity's output into my custom activity by specifying it via Liquid syntax:

![image](https://github.com/elsa-workflows/elsa-core/assets/1834711/e892a118-652b-42e9-85fd-5688ba586f7d)

I have an `[ActivityInput]` defined like this on my custom activity:

![MicrosoftTeams-image](https://github.com/elsa-workflows/elsa-core/assets/1834711/14fdeb09-712b-4f60-9db3-155e92df2470)

I can get it successfully inside the activity via: 
```csharp
var data = ((HttpRequestModel)context.Input).Body as IntacctProjectTriggerEvent;
```

So the **HttpEndpoint** activity does de-serialize it successfully. I'd really prefer to get it to populate the model directly on the input though. Currently it always comes through as `null` for that input.

**Comment by sfmskywalker** [comment, marked as answer]
Hi Adam!

Thank you for providing a detailed explanation of your setup. Instead of using Liquid, please try and use JavaScript syntax instead. When using the Liquid syntax, the result of the expression will always be a string, and I don't think Elsa 2 will attempt to deserialize the input string into your input type.

You can try the following JS expression instead:

```javascript
activities.endpoint.Output().Body
```

This should help you get the desired output directly into your custom activity's input. Let me know if you face any other issues!

**Reply by adamfisher** [reply]
That worked! Thank you! I actually have two different endpoints so I switch between them but adding those parenthesis on `Output()` is what I was missing.

```javascript
var endpoint = activities.Endpoint.Output();
var realTimeEndpoint = activities.RealtimeEndpoint.Output();

if(endpoint != null) {
    return endpoint.Body;
}
else {
    return realTimeEndpoint.Body;
}
```
